docker-multinode/common.sh

```diff
@@ -0,0 +1,437 @@
+    lsb_dist='fedora'
+  elif [[ -r /etc/os-release ]]; then
+    lsb_dist="$(. /etc/os-release && echo "$ID")"
+  elif kube::helpers::command_exists systemctl; then
+    lsb_dist='systemd'
+  fi
+
+  lsb_dist="$(echo ${lsb_dist} | tr '[:upper:]' '[:lower:]')"
+
+  case "${lsb_dist}" in
+      amzn|centos|debian|ubuntu|systemd)
+          ;;
+      *)
+          kube::log::error "Error: We currently only support ubuntu|debian|amzn|centos|systemd."
+          exit 1
+          ;;
+  esac
+
+  kube::log::status "Detected OS: ${lsb_dist}"
+}
+
+# Start a docker bootstrap for running etcd and flannel
+kube::multinode::bootstrap_daemon() {
+
+  kube::log::status "Launching docker bootstrap..."
+
+  docker daemon \
+    -H ${BOOTSTRAP_DOCKER_SOCK} \
+    -p /var/run/docker-bootstrap.pid \
+    --iptables=false \
+    --ip-masq=false \
+    --bridge=none \
+    --graph=/var/lib/docker-bootstrap \
+      2> /var/log/docker-bootstrap.log \
+      1> /dev/null &
+
+  # Wait for docker bootstrap to start by "docker ps"-ing every second
+  local BOOTSTRAP_SECONDS=0
+  while [[ $(docker -H ${BOOTSTRAP_DOCKER_SOCK} ps 2>&1 1>/dev/null; echo $?) != 0 ]]; do
+    ((BOOTSTRAP_SECONDS++))
+    if [[ ${BOOTSTRAP_SECONDS} == ${TIMEOUT_FOR_SERVICES} ]]; then
+      kube::log::error "docker bootstrap failed to start. Exiting..."
+      exit
+    fi
+    sleep 1
+  done
+}
+
+# Start etcd on the master node
+kube::multinode::start_etcd() {
+
+  kube::log::status "Launching etcd..."
+  
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run -d \
+    --restart=${RESTART_POLICY} \
+    --net=host \
+    gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+    /usr/local/bin/etcd \
+      --listen-client-urls=http://127.0.0.1:4001,http://${MASTER_IP}:4001 \
+      --advertise-client-urls=http://${MASTER_IP}:4001 \
+      --data-dir=/var/etcd/data
+
+  # Wait for etcd to come up
+  kube::util::wait_for_url "http://localhost:4001/v2/machines" "etcd" 0.25 80
+
+  # Set flannel net config
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run \
+      --net=host \
+      gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+      etcdctl \
+      set /coreos.com/network/config \
+          "{ \"Network\": \"${FLANNEL_NETWORK}\", \"Backend\": {\"Type\": \"${FLANNEL_BACKEND}\"}}"
+
+  sleep 2
+}
+
+# Start flannel in docker bootstrap, both for master and worker
+kube::multinode::start_flannel() {
+
+  kube::log::status "Launching flannel..."
+
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run \
+    --restart=${RESTART_POLICY} \
+    -d \
+    --net=host \
+    --privileged \
+    -v /dev/net:/dev/net \
+    -v ${FLANNEL_SUBNET_TMPDIR}:/run/flannel \
+    quay.io/coreos/flannel:${FLANNEL_VERSION} \
+    /opt/bin/flanneld \
+      --etcd-endpoints=http://${MASTER_IP}:4001 \
+      --ip-masq="${FLANNEL_IPMASQ}" \
+      --iface="${FLANNEL_IFACE}"
+
+  # Wait for the flannel subnet.env file to be created instead of a timeout. This is faster and more reliable
+  local FLANNEL_SECONDS=0
```

Comment:
You can use $SECONDS as an internal shell variable.
It is automatically increased after second. You just need reset this variable before while loop: SECONDS=0